backend/app/routers/authorization.py

```diff
@@ -318,3 +320,65 @@ async def remove_user_role(
             raise HTTPException(status_code=404, detail=f"User {username} not found")
     else:
         raise HTTPException(status_code=404, detail=f"Dataset {dataset_id} not found")
+
+
+@router.get("/datasets/{dataset_id}/users_and_roles", response_model=List[UserAndRole])
+async def get_dataset_users_and_roles(
```

Comment:
That was part of my thinking. Users and Groups are also different in terms of their fields and how they are used for permissions, so I couldn't figure out 1 solution for both. For users you add their email to the dataset with a role, but for groups you add the actual uuid to the dataset with the role and not the group name. Using 1 object looked like it would require a bunch of extra checks for whether it's a group or a user, so I just did 2 separate object types with 2 endpoints.